src/Auth.cpp

```diff
@@ -111,14 +112,25 @@ STATUS CredentialProvider::getStreamingTokenHandler(UINT64 custom_data, PCHAR st
     freeAwsCredentials(&this_obj->security_token_);
 
     // Store the buffer so we can release it at the end
-    createAwsCredentials((PCHAR) access_key.c_str(), access_key_len, (PCHAR) secret_key.c_str(), secret_key_len,
-                         (PCHAR) session_token.c_str(), session_token_len, expiration, &this_obj->security_token_);
+    if(IS_EMPTY_STRING(session_token.c_str())) {
```

Comment:
Q: is 0 length access key secret key handled here or is that handled somewhere else

Reply:
0 is an allowed value. The API calculates the length anyways if it is supplied as 0.